lvgl_helpers.h

```diff
@@ -34,48 +56,64 @@ extern "C" {
  * color format being used, for RGB565 each pixel needs 2 bytes.
  * When using the mono theme, the display pixels can be represented in one bit,
  * so the buffer size can be divided by 8, e.g. see SSD1306 display size. */
+
+#define TFT_DISPLAY_BUFFER_SIZE_IN_BITS (LV_HOR_RES_MAX * 40 * 3 * 8)
+#define TFT_DISPLAY_BUFFER_SIZE (LV_HOR_RES_MAX * 40)
+
+#if CONFIG_IDF_TARGET_ESP32S3 
+    #define DMA_MAX_BIT_LENGHT (1<<18) // according with SPI_LL_DMA_MAX_BIT_LEN in spi_ll.h
+#else
+    #define DMA_MAX_BIT_LENGHT (1<<24) // according with SPI_LL_DMA_MAX_BIT_LEN in spi_ll.h
+#endif
+
+#define TFT_DISPLAY_BUFFER_SIZE_OVERFLOW_PROTECTION ((TFT_DISPLAY_BUFFER_SIZE_IN_BITS>DMA_MAX_BIT_LENGHT)? (((DMA_MAX_BIT_LENGHT-1000)/8)/3) :TFT_DISPLAY_BUFFER_SIZE)
```

Comment:
I don't like this approach here. We can always break one larger buffer into multiple DMA requests. Limit the DISP_BUF_SIZE is not the best option. In SPI code, we can break one request into multiple without every driver to handle it.